### DISCUSSION TITLE
Custom Filter Lists getting randomly deleted

### DISCUSSION THREAD

**Discussion by click-click** [uBlockOrigin/uBlock-issues#3035] [Q&A]
Ever since upgrading to 154.0 on Firefox portable, my custom filter lists keep getting randomly deleted. I've restored settings several times now, but this should not be happening. I usually notice the filter list disappearance problem when visiting frequented sites where things are no longer being blocked.  Does anyone know what could be causing this to happen or know how to debug this problem?

**Comment by garry-ut99** [comment]
I only know this bug, but as far I see, it was happening only in `1.54.1b4` :
- https://github.com/uBlockOrigin/uBlock-issues/issues/3003

**Reply by Yuki2718** [reply]
https://github.com/uBlockOrigin/uBlock-issues/issues/2335

**Reply by garry-ut99** [reply]
https://www.reddit.com/r/uBlockOrigin/comments/18k1ee2/custom_filter_lists_deleted_after_installing/

**Comment by gorhill** [comment]
> my custom filter lists

Do you mean imported lists in _"Filter lists"_ pane or custom filters from _"My filters"_ pane?

**Reply by click-click** [reply]
imported lists in "Filter lists" pane.
 >Custom 8/14

**Reply by gorhill** [reply]
Are they removed or unselected?

You have an example of such list we can test on our side?

**Reply by click-click** [reply]
They are removed. I have to restore my settings. Should I attach my settings backup for you to test?

Note: this happens randomly after running with 154. 

Also, when I updated from 153 to 154 via firefox (available updates after check for updates), I selected update now instead of update after restart and that causes the filters to be deleted after restarting Firefox.

**Reply by u-RraaLL** [reply]
> They are removed.
>> Custom 8/14

This shows disabled, not removed. Or is that not the actual copy of the field and you just wrote how many got removed?

Does it happen to specific lists? As in the same list over and over? Or a different list each time?
If it's the same list every time, give us an example of one, like gorhill asked.

**Reply by click-click** [reply]
Yes, counts before things got removed. No specific lists. All Custom filter lists.

**Reply by gorhill** [reply]
I suggest you try dev build, and if issue still persist, I guess this is related to the fact you are using Firefox Portable, which from what I read saves things to storage differently.
